docker-app/qfieldcloud/core/middleware/requests.py

```diff
@@ -17,15 +17,20 @@ def attach_keys(get_response):
     """
 
     def middleware(request):
+        try:
+            sentry_max_size = int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)
+        except (ValueError, TypeError):
+            logger.error(
+                "Invalid SENTRY_REQUEST_MAX_SIZE_TO_SEND value. Defaulting to 0."
+            )
+            sentry_max_size = 0
+
         # add a copy of the request body to the request
         if (
             settings.SENTRY_DSN
             and request.method == "POST"
             and "Content-Length" in request.headers
-            and (
-                int(request.headers["Content-Length"])
-                < config.SENTRY_REQUEST_MAX_SIZE_TO_SEND
-            )
+            and (int(request.headers["Content-Length"]) < sentry_max_size)
```

Comment:
```suggestion
            and (
                int(request.headers["Content-Length"])
                < int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)
            )
```
Even if you insist keeping the code in the previous hunk, this is the preferable change, so it is clear what is clear where `SENTRY_REQUEST_MAX_SIZE_TO_SEND` is coming from.

If I were you, I would drop the previous hunk and keep stick the PR only to the change proposed here.

Reply:
I would suggest a more defensive way such as

```
        try:
            if (
            settings.SENTRY_DSN
            and request.method == "POST"
            and "Content-Length" in request.headers
            and (int(request.headers["Content-Length"]) < int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND))
            ):
                logger.info("Making a temporary copy for request body.")

                input_stream = io.BytesIO(request.body)
                output_stream = io.BytesIO()
                shutil.copyfileobj(input_stream, output_stream)
                request.body_stream = output_stream
        except (ValueError, TypeError) as e:
            logger.error(
                f"Invalid SENTRY_REQUEST_MAX_SIZE_TO_SEND value: {config.SENTRY_REQUEST_MAX_SIZE_TO_SEND} "
                f"({type(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND).__name__}). Error: {e}"
            )
            raise 
```

Reply:
I think this reduces the readability of the code. We have two paths for this error:

1) fix the root cause of the error, because constance is aware this must be an int but somehow it isn't.
2) just assume constance is temporarily failng to give the value as an int. If the value is not parsable int at all, then we need to fix 1) for sure.

Reply:
Just tried locally:
```
In [3]: from constance import config

In [4]: config.SENTRY_REQUEST_MAX_SIZE_TO_SEND
Out[4]: 1000

In [5]: type(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)
Out[5]: int
```

Reply:
Same on prod (different value)

Reply:
Let's go with option 2 while staying alert for any errors. In case the value is not a parsable int at all, we will dig into it :)

> I think this reduces the readability of the code. We have two paths for this error:
> 
> 1. fix the root cause of the error, because constance is aware this must be an int but somehow it isn't.
> 2. just assume constance is temporarily failng to give the value as an int. If the value is not parsable int at all, then we need to fix 1) for sure.